source/_includes/common-tasks/flashing_m1s_otg.md

```diff
@@ -0,0 +1,41 @@
+Instead the M1S has a build-in 64GB eMMC soldered directly on the board that holds a bootloader by default. This guide will show you how to flash HAOS to the build-in eMMC.
+
+<ins>**Warning:</ins> Installing HAOS replaces the firmware and SPL on the eMMC with the mainline version provided by HAOS. As a result, it is not possible to use the SD card with the EMMC2UMS image anymore, because the mainline SPL is not compatible with U-Boot in the EMMC2UMS image at this time (February 2024). This does not pose any problem for standard use, just makes it more complicated in case you want to return to the Hardkernel-provided OS (see HK Recovery).**
+
+### What you will need
+
+To flash your eMMC using OTG-USB, you will need the following items:
+
+- A small SD card
+- Another computer
+- USB 2.0 to micro-USB cable
+- the special Hardkernel eMMC-to-USB-mass-storage image
+
+### Boot into mass-storage mode
+(These steps are identical to the official [Hardkernel wiki](https://wiki.odroid.com/odroid-m1s/getting_started/os_installation_guide?redirect=1#install_over_usb_from_pc) page.)
+
+1. Download [ODROID-M1S_EMMC2UMS.img](https://dn.odroid.com/RK3566/ODROID-M1S/Installer/ODROID-M1S_EMMC2UMS.img)
+2. Use [balena Etcher](https://www.balena.io/etcher/) or another tool to flash the UMS utility onto an SD card.
+   - use **Flash from file**. **Flash from URL** does not work on all systems.
+      (balena Etcher will complain that something went wring during flashin. You can ignore this message)
+3. Plug-in that SD card to your ODROID-M1S and boot it.
+
+### Flashing Home Assistant M1S
+1. Download the latest stable version of Home Assistant OS for the [ODROID-M1S](https://github.com/home-assistant/operating-system/releases/download/{{site.data.version_data.hassos['odroid-m1s']}}/haos_odroid-m1s-{{site.data.version_data.hassos['odroid-m1s']}}.img.xz).
+2. Apart from the HAOS image to flash (M1S instead of N2+ version), you can follow the N2+ step-by-step flashing guide [HERE](/common-tasks/os/#flashing-home-assistant).
+
+
+#### _HK Recovery_
+If you want to restore your M1S back into Hardkernel's initial state, you will have to restore the HK's bootloader.
+A reliable way of reflashing the eMMC in this case is to use HAOS booted from an SD card. To do that, insert the SD card with HAOS to the micro SD slot and plug the board in. Once the device boots to the HA CLI, enter `login` to enter the root shell and use `curl` to download an image and `dd` it to the eMMC block device:
+
+```sh
+curl https://dn.odroid.com/RK3566/ODROID-M1S/Installer/ODROID-M1S_EMMC2UMS.img | dd of=/dev/mmcblk0
+```
+
+This way the device will start in the UMS mode on the next boot with the SD card removed. Alternatively you can use the [Hardkernel installer image](https://wiki.odroid.com/odroid-m1s/getting_started/os_installation_guide#user_installer) directly instead of the EMMC2UMS image.
```

Comment:
Thank you for reviewing that.


> I don't think this should be in the user documentation. While convenient during development and for those know what they do, for lots of users this is probably confusing. We have it in the more power-user/developer oriented documentation in the OS repo (https://github.com/home-assistant/operating-system/blob/e60a2e4b68c061b920347b9394e5b95696d306cd/Documentation/boards/hardkernel/odroid-m1s.md#emmc).

I personally am fine with keeping it there, because even as a less experienced user it would make me feel just comfortable if I directly see, that there is a "way back" from what I have been warned before. Or the other way: if I get warned about something and I don't see how/if I could revert that, this would make me kind of nervous.
If course I can remove it again and maybe just refer to the power-user guide if you or the majority asks me to do so. But my guts are fine with keeping it there ;) .

Reply:
Sorry, my comment was a bit rushed and unclear.

> I personally am fine with keeping it there, because even as a less experienced user it would make me feel just comfortable if I directly see, that there is a "way back" from what I have been warned before.

I agree with that.

However, the method described here is somewhat advanced. Folks will type this command into the web terminal, where access to  `/dev/mmcblk0` is not possible. Every documentation needs to be maintained, if users are unclear issues being handled etc.  Sometimes more is more, but sometimes less is more :sweat_smile: 

There is a more generic way of recovering, which doesn't even need shell access. and we should advice to use this by default. So I'd suggest to just document the Hardkernel installer image method, and link to the external wiki in this user documentation.

Reply:
> There is a more generic way of recovering, which doesn't even need terminal access. and we should advice to use this by default. So I'd suggest to just document the Hardkernel installer image method, and link to the external wiki in this user documentation.

The Hardkernel images do not boot with mainline u-boot installed on eMMC though. That was the entire reason for the workaround in the first place. The only other option is shorting out maskROM pads to boot hardkernel images.

Reply:
@darkxst so then this sentence is wrong?

> Alternatively you can use the [Hardkernel installer image](https://wiki.odroid.com/odroid-m1s/getting_started/os_installation_guide#user_installer) directly instead of the EMMC2UMS image.

Reply:
Ah I understand, it is probably meant to be flashed using dd directly, similarly to ODROID-M1S_EMMC2UMS.img.

A few problems though: The command would be different as that image is xz compresseed. And this is definitely is not safe/guarnateed to work, as the image is larger and would overwrite the squashfs rootfs partition of HAOS. :cry:

Reply:
Oh, nevermind, the commands are meant to be run from a HAOS running on a SD card. Yeah ok, then both methods should work and are safe.

The Hardkernel installer image can't be installed directly though because HAOS is missing unxz :cry: 

Also, I'd say that the command should work safely when booted from eMMC. The image is only 3MB, so this does not alter any of the mounted partitions. Having two HAOS installation on separate partition usually ends up in a bit chaotic situations since we rely on static partition UUID to find rootfs partitions etc.

I'll make a suggestion against the updated PR.

Reply:
> The only other option is shorting out maskROM pads to boot hardkernel images.

One other option is to make our own UMS tool with mainline uboot. It is just uboot with UMS enabled in that Hardkernel image.

Reply:
I added the suggestion of @agners (see https://github.com/home-assistant/home-assistant.io/pull/31540/commits/86671fc6a59eac5d2e8fde9e6a9383b85df4fb25).
Can I clase this one as resolved?

Reply:
@darkxst 
> The only other option is shorting out maskROM pads to boot hardkernel images.

But IIUC that should get you to the Rockchip download mode, which is even more involved procedure :grimacing:

> yes that is it, sairon added that comment to the docs, but guess he didnt test it

Yes, indeed I did not :sweat_smile: A simple piping through `xz -dc` should solve it, I guess I should be more explicit in the OS docs.

Reply:
> But IIUC that should get you to the Rockchip download mode, which is even more involved procedure 😬

Only if it can't boot from SD card, will it fallback to RockUSB.

Reply:
Ah, you're right. I got the impression it doesn't boot from the SD from [this post](https://forum.odroid.com/viewtopic.php?p=377662&sid=3fc9d7429603736e8dd7c2ca4c1ff904#p377662) but I just tested it behaves the way you say :+1: